pkg/vault/contracts/BatchRouter.sol

```diff
@@ -671,6 +671,317 @@ contract BatchRouter is IBatchRouter, BatchRouterStorage, RouterCommon, Reentran
+        uint256[] memory maxWrappedAmountsIn = new uint256[](erc4626PoolTokens.length);
+
+        for (uint256 i = 0; i < erc4626PoolTokens.length; ++i) {
+            maxWrappedAmountsIn[i] = IERC4626(address(erc4626PoolTokens[i])).previewDeposit(params.maxAmountsIn[i]);
+        }
+
+        // Add wrapped amounts to the boosted pool
+        (amountsIn, bptAmountOut, returnData) = _vault.addLiquidity(
+            AddLiquidityParams({
+                pool: params.pool,
+                to: params.sender,
+                maxAmountsIn: maxWrappedAmountsIn,
+                minBptAmountOut: params.minBptAmountOut,
+                kind: params.kind,
+                userData: params.userData
+            })
+        );
+
+        _wrapToken(params, erc4626PoolTokens, amountsIn, SwapKind.EXACT_OUT, type(uint256).max);
+    }
+
+    function removeLiquidityERC4626PoolProportionalHook(
+        RemoveLiquidityHookParams calldata params
+    )
+        external
+        nonReentrant
+        onlyVault
+        returns (uint256 bptAmountIn, uint256[] memory amountsOut, bytes memory returnData)
+    {
+        IERC20[] memory erc4626PoolTokens = _vault.getPoolTokens(params.pool);
+        uint256[] memory minWrappedAmountsIn = new uint256[](erc4626PoolTokens.length);
+        uint256[] memory underlyingAmountsOut = new uint256[](erc4626PoolTokens.length);
+
+        for (uint256 i = 0; i < erc4626PoolTokens.length; ++i) {
+            minWrappedAmountsIn[i] = IERC4626(address(erc4626PoolTokens[i])).previewWithdraw(params.minAmountsOut[i]);
+        }
+
+        (bptAmountIn, amountsOut, returnData) = _vault.removeLiquidity(
+            RemoveLiquidityParams({
+                pool: params.pool,
+                from: params.sender,
+                maxBptAmountIn: params.maxBptAmountIn,
+                minAmountsOut: minWrappedAmountsIn,
+                kind: params.kind,
+                userData: params.userData
+            })
+        );
+
+        bool isStaticCall = EVMCallModeHelpers.isStaticCall();
+        for (uint256 i = 0; i < erc4626PoolTokens.length; ++i) {
+            IERC4626 wrappedToken = IERC4626(address(erc4626PoolTokens[i]));
+            IERC20 underlyingToken = IERC20(wrappedToken.asset());
+
+            // erc4626BufferWrapOrUnwrap will fail if the wrapper wasn't ERC4626
+            (, , underlyingAmountsOut[i]) = _vault.erc4626BufferWrapOrUnwrap(
+                BufferWrapOrUnwrapParams({
+                    kind: SwapKind.EXACT_IN,
+                    direction: WrappingDirection.UNWRAP,
+                    wrappedToken: wrappedToken,
+                    amountGivenRaw: amountsOut[i],
+                    limitRaw: 0, // We don't apply the limit here
+                    userData: params.userData
+                })
+            );
+
+            if (isStaticCall == false) {
+                _sendTokenOut(params.sender, underlyingToken, underlyingAmountsOut[i], params.wethIsEth);
+            }
```

Comment:
Another option would be to do this outside the hook, but maybe it's just fine like this.